Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Common.hs

```diff
@@ -207,23 +207,23 @@ mkBppUri merchantId =
   asks (.nwAddress)
     <&> #baseUrlPath %~ (<> "/" <> T.unpack merchantId)
 
-castVariant :: Variant.Variant -> (Text, Text)
-castVariant Variant.SEDAN = (show Enums.CAB, "SEDAN")
-castVariant Variant.HATCHBACK = (show Enums.CAB, "HATCHBACK")
-castVariant Variant.SUV = (show Enums.CAB, "SUV")
-castVariant Variant.AUTO_RICKSHAW = (show Enums.AUTO_RICKSHAW, "AUTO_RICKSHAW")
-castVariant Variant.TAXI = (show Enums.CAB, "TAXI")
-castVariant Variant.TAXI_PLUS = (show Enums.CAB, "TAXI_PLUS")
-castVariant Variant.PREMIUM_SEDAN = (show Enums.CAB, "PREMIUM_SEDAN")
-castVariant Variant.BLACK = (show Enums.CAB, "BLACK")
-castVariant Variant.BLACK_XL = (show Enums.CAB, "BLACK_XL")
-castVariant Variant.BIKE = (show Enums.MOTORCYCLE, "BIKE")
-castVariant Variant.AMBULANCE_TAXI = (show Enums.AMBULANCE, "AMBULANCE_TAXI")
-castVariant Variant.AMBULANCE_TAXI_OXY = (show Enums.AMBULANCE, "AMBULANCE_TAXI_OXY")
-castVariant Variant.AMBULANCE_AC = (show Enums.AMBULANCE, "AMBULANCE_AC")
-castVariant Variant.AMBULANCE_AC_OXY = (show Enums.AMBULANCE, "AMBULANCE_AC_OXY")
-castVariant Variant.AMBULANCE_VENTILATOR = (show Enums.AMBULANCE, "AMBULANCE_VENTILATOR")
-castVariant Variant.SUV_PLUS = (show Enums.CAB, "SUV_PLUS")
+castVariant :: Variant.Variant -> Bool -> (Maybe Text, Maybe Text)
+castVariant Variant.SEDAN _ = (Just $ show Enums.CAB, Just "SEDAN")
```

Comment:
can you simplify this?
top level you can check for `isValueAddNP == false` then handle special case, otherwise default logic

Reply:
Also you can use guard to check for base cases, and for all others you can use `show`